documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+        * [Error Response](#error-response-4)
+* [The Scope Parameter](#the-scope-parameter)
+* [ID Token](#id-token)
+* [Client Authentication](#client-authentication)
+* [References](#references)
+* [Appendix A. RSA Key Used in Examples](#appendix-a-rsa-key-used-in-examples)
+    * [Appendix A.1. Authorization Server RSA Key](#appendix-a1-authorization-server-rsa-key)
+    * [Appendix A.2. Client RSA Key](#appendix-a2-client-rsa-key)
+<!-- TOC -->
+
+
+## Introduction
+
+Telco network capabilities exposed through APIs provide great value to customers. By simplifying the complexity of telco networks with APIs and making the APIs available across telco networks and countries, CAMARA enables easy and seamless access. To ensure seamless interoperability and increased security, this technical document introduces the OpenID Connect (OIDC) profile, which is specifically tailored for CAMARA APIs.
+
+OpenID Connect is a widely adopted standard for user authentication and authorization in the context of APIs. This document builds on the basic principles of OIDC and outlines CAMARA-specific guidelines to address CAMARA APIs scenarios and use cases.
+
+The CAMARA OIDC profile includes three different authorization mechanisms: Authorization Code Flow, Client Initiated Backchannel Authentication (CIBA), and Client Credentials. Each method is detailed, providing comprehensive insight into the authorization and token request processes, mandatory parameters, successful response content or potential error scenarios.
+
+Key areas covered include:
+
+* Authorization flows:
+
+    * Authorization Code Flow.
+    * Client Initiated Back Channel Authentication (CIBA).
+    * Client Credentials.
+
+* Scope Parameter. Definition of scope parameter format applicable to CAMARA APIs.
+
+* ID Token Content. CAMARA specific details about the content of the Id token.
+
+* Client Authentication. Specifications for client authentication within the CAMARA OIDC profile.
+
+By encapsulating these elements within the CAMARA OIDC profile, this document aims to provide a comprehensive guide for developers and operators, ensuring consistent implementation and adherence to standardized security measures across the CAMARA ecosystem. The defined OIDC profile not only facilitates the integration process, but also serves as a basic framework for developers wishing to leverage the CAMARA APIs while maintaining security and interoperability.
+
+
+## Audience
+
+The target audience for this document is the service/technical departments of operators exposing network functions via standard CAMARA APIs. And applications or client systems that consume CAMARA standard APIs to make use of the operator's network capabilities.
+
+
+## Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
```

Comment:
## Camara security and interoperability best practises

### Network Layer Protections

Regarding network layer protections Camara follows [FAPI 2.0 Security](https://openid.bitbucket.io/fapi/fapi-2_0-security-profile.html#name-network-layer-protections)

---

### Requirements for all endpoints

All TLS connections between web browsers, clients, authorization
servers, and resource servers shall be protected against network
attackers. To this end, clients, authorization servers, and resource
servers

 1. shall only offer TLS protected endpoints and shall establish connections 
    to other servers using TLS;
 1. shall set up TLS connections using TLS version 1.2 or later;
 2. when using TLS 1.2, shall follow the recommendations for Secure Use of Transport Layer Security in [@!BCP195];
 3. should use DNSSEC to protect against DNS spoofing attacks that can lead to
    the issuance of rogue domain-validated TLS certificates; and
 4. shall perform a TLS server certificate check, as per [@!RFC6125].

**NOTE**: Even if an endpoint uses only organization validated (OV) or extended 
validation (EV) TLS certificates, rogue domain-validated certificates can be used 
to impersonate the endpoint and conduct man-in-the-middle attacks. CAA records
[@!RFC8659] can help to mitigate this risk.

### Requirements for endpoints not used by web browsers

For server-to-server communication endpoints that are not used by web
browsers, the following requirements apply:

 1. When using TLS 1.2, servers shall only permit the cipher suites listed in (#tls-12-ciphers).
 2. When using TLS 1.2, clients should only permit the cipher suites listed in (#tls-12-ciphers).
 3. When using the `TLS_DHE_RSA_WITH_AES_128_GCM_SHA256` or `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384` cipher suites, 
 key lengths of at least 2048 bits are required.

#### TLS 1.2 permitted cipher suites {#tls-12-ciphers}

For TLS 1.2, only the following cipher suites shall be used:

  * `TLS_DHE_RSA_WITH_AES_128_GCM_SHA256`
  * `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`
  * `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384`
  * `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384`

### Requirements for endpoints used by web browsers

For endpoints that are used by web browsers, the following additional
requirements apply:

  1. Servers shall use methods to ensure that connections cannot be
     downgraded using TLS Stripping attacks. A preloaded [@preload] HTTP
     Strict Transport Security policy [@!RFC6797] can be used for this
     purpose.
  2. When using TLS 1.2, servers shall only use cipher suites allowed in
     [@!BCP195].